File: frontend/src/types/adminStats.ts
// Types for GET /admin/registered-user-count, matching
// docs/API_CONTRACTS.md "GET /admin/registered-user-count" -- a live
// Cognito-backed count, separate from GET /admin/overview (which stays a
// local-DB aggregate; see that section's own "registered_user_count is
// always null on THIS endpoint" note for why the two are split).

export interface RegisteredUserCount {
  count: number;
  /** Always "registered_user" today -- see the backend schema's own comment. */
  group: string;
}

File: backend/app/schemas/admin_stats.py
"""`GET /admin/registered-user-count` — see docs/API_CONTRACTS.md "Admin
(bulk operations)" section. Its own schema module (not folded into
`admin_notifications.py`) since it's an unrelated data source (Cognito
group membership, not a local-DB aggregate) and a companion admin
analytics/overview page is expected to grow this file with sibling stats
later.
"""
from __future__ import annotations

from pydantic import BaseModel


class RegisteredUserCountResponse(BaseModel):
    count: int
    # Named explicitly rather than assumed, since "registered users" is
    # ambiguous between "everyone in the pool" and "the registered_user
    # role" — see the router docstring for the reasoning.
    group: str = "registered_user"
